hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -334,6 +335,13 @@ def from_typing_type(thing):
     if get_origin(thing) == tuple or isinstance(
         thing, getattr(typing, "TupleMeta", ())
     ):
```

Comment:
It turns out that the `TupleMeta` case hasn't been necessary since we dropped support for Python 3.6, which means we can move this logic to a new `@register`'d `resolve_tuple()` function below.  That will treat `tuple` as if it's any other generic type, and should simplify things considerably!

Reply:
Currently, I'm just pre-registering `lambda _: NotImplmented`, rather than going the route i mentioned earlier of pre-registering the default tuple resolution logic. Are you suggesting I go back to that option? And you're saying that some logic in `from_typing_type` will handle tuples, if we delete the tuple special casing from `from_typing_type`?